### PR TITLE
Hlapi trivial encrypts

### PR DESCRIPTION
**Pull request by tmontaigu**
closes zama-ai/tfhe-rs-internal#93

### CODE REVIEW

**Comment by tmontaigu** [comment]
@slab-ci cpu_integer_test

File: tfhe/src/high_level_api/integers/types/base.rs [hunk 1]
```diff
@@ -166,6 +168,47 @@ where
...
+    fn try_encrypt_trivial(value: T) -> Result<Self, Self::Error> {
+        let value = value.into();
+        let id = P::Id::default();
+        let ciphertext = id.with_global(|key| match key.pbs_order {
+            crate::shortint::PBSOrder::KeyswitchBootstrap => {
+                RadixCiphertextDyn::Big(key.inner.create_trivial_radix(value, key.num_block))
+            }
+            crate::shortint::PBSOrder::BootstrapKeyswitch => {
+                RadixCiphertextDyn::Big(key.inner.create_trivial_radix(value, key.num_block))
+            }
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/types/base.rs:191]: ::Small ?

**Review by IceTDrinker — COMMENTED** [review]
Maybe just a copy paste error, other wise looks good I would say !

File: tfhe/src/high_level_api/integers/types/base.rs (hunk 1 above)
Comment (tmontaigu) [reply tfhe/src/high_level_api/integers/types/base.rs:191]: yes nicely caught

Comment (IceTDrinker) [reply tfhe/src/high_level_api/integers/types/base.rs:191]: easier when the PRs are not 4000 lines long 🤣 (sorry about mine btw, those were huge...)

File: tfhe/src/high_level_api/integers/types/base.rs [hunk 2]
```diff
@@ -166,6 +168,47 @@ where
...
+    fn try_encrypt_trivial(value: T) -> Result<Self, Self::Error> {
+        let value = value.into();
+        let id = P::Id::default();
+        let ciphertext = id.with_global(|key| match key.pbs_order {
+            crate::shortint::PBSOrder::KeyswitchBootstrap => {
+                RadixCiphertextDyn::Big(key.inner.create_trivial_radix(value, key.num_block))
+            }
+            crate::shortint::PBSOrder::BootstrapKeyswitch => {
+                RadixCiphertextDyn::Small(key.inner.create_trivial_radix(value, key.num_block))
+            }
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/types/base.rs:191]: can we maybe add a small test somewhere to check we are getting the right CT type for trivial encryption ? Or maybe just an assert in the u128 and u256 tests you wrote to check we are getting big/small as expected?

Comment (tmontaigu) [reply tfhe/src/high_level_api/integers/types/base.rs:191]: I'll add one

File: tfhe/src/high_level_api/integers/types/base.rs (hunk 1 above)
Comment (tmontaigu) [reply tfhe/src/high_level_api/integers/types/base.rs]: 😏

File: tfhe/src/high_level_api/integers/types/base.rs (hunk 2 above)
Comment (tmontaigu) [reply tfhe/src/high_level_api/integers/types/base.rs:191]: added asserts

**Review by IceTDrinker — APPROVED** [review]
Thanks for the added asserts!

**Comment by IceTDrinker** [comment]
does this require a cherry-pick in 0.2.x ?

**Pull request merged**